src/main/java/bio/terra/common/retry/CompositeBackOffPolicy.java

```diff
@@ -0,0 +1,77 @@
+package bio.terra.common.retry;
+
+import bio.terra.common.db.DatabaseRetryUtils;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.classify.BinaryExceptionClassifier;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.BackOffContext;
+import org.springframework.retry.backoff.BackOffInterruptedException;
+import org.springframework.retry.backoff.BackOffPolicy;
+
+/**
+ * BackOffPolicy that delegates to a number of nested BackOffPolicies. The BackOffPolicy chosen is
+ * determined by the exception being handled. If more than one policy matches, the first is used.
+ */
+public class CompositeBackOffPolicy implements BackOffPolicy {
+  private static final Logger logger = LoggerFactory.getLogger(DatabaseRetryUtils.class);
+
+  // LinkedHashMap to ensure consistent behavior when an exception matches more than 1 policy
+  private final LinkedHashMap<BinaryExceptionClassifier, BackOffPolicy> backOffPoliciesByClassifier;
+
+  public CompositeBackOffPolicy(
+      LinkedHashMap<BinaryExceptionClassifier, BackOffPolicy> backOffPoliciesByClassifier) {
+    this.backOffPoliciesByClassifier = backOffPoliciesByClassifier;
+  }
+
+  @Override
+  public BackOffContext start(RetryContext context) {
+    Map<BinaryExceptionClassifier, BackOffContext> backOffContextMap = new HashMap<>();
```

Comment:
Why is this a map? You don't use the map functionality. It could be an array of Pair.

Reply:
correct